src/rpdk/core/contract/resource_client.py

```diff
@@ -288,22 +288,36 @@ def assert_failed(status, response):
         return error_code
 
     @staticmethod
+    # pylint: disable=R0913
     def make_request(
         desired_resource_state,
         previous_resource_state,
         region,
         account,
         partition,
+        action,
+        creds,
+        client_token,
+        callback_context=None,
         **kwargs
     ):
         return {
-            "desiredResourceState": desired_resource_state,
-            "previousResourceState": previous_resource_state,
-            "logicalResourceIdentifier": None,
-            **kwargs,
+            "requestData": {
+                "callerCredentials": creds,
+                "providerCredentials": creds,
+                "resourceProperties": desired_resource_state,
+                "previousResourceProperties": previous_resource_state,
+                "logicalResourceIdentifier": None,
```

Comment:
let's give a logicalResourceIdentifier. Some resources needs it to generate name etc.